backend/src/scripts/whitelisting/whitelist_users.ts

```diff
@@ -0,0 +1,246 @@
+/* eslint-disable no-console */
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+require('dotenv').config()
+require('module-alias/register')
+
+import sequelizeLoader from '@core/loaders/sequelize.loader'
+import { Domain, User } from '@core/models'
+import { Op } from 'sequelize'
+import csv from 'csvtojson'
+import { UserExperimental } from '@core/models/user/user-experimental'
+import { ChannelType } from '@core/constants'
+import { GovsgTemplate, GovsgTemplatesAccess } from '@govsg/models'
+import {
+  ALL_TEMPLATE_ACCESS,
+  MCI_TEMPLATE_ACCESS,
+  OGP_TEMPLATE_ACCESS,
+  SECTION_DIVIDER,
+} from './constants'
+
+type RawOfficerData = {
+  name: string
+  email: string
+  designation: string
+  agency: string
+  templates: string[]
+  id?: number
+}
+
+/**
+ * Receives a csv and whitelists users to use the govsg feature.
+ * In addition to giving users default access for the different template groups (OGP, MCI, ALL),
+ * The user can be given adhoc access by passing in a list of template labels separated by ';' in the 5th column of the csv.
+ * Example: template1;template2;template3
+ */
+void (async function main() {
```

Comment:
Just to confirm my understanding: This script should be used when we specify the 5th column in the whitelist csv.

As for `update_whitelist.ts`, it's typically used when:

- The 5th column is not specified and now we want to give the right access.
- For past users who never went through the `whitelist_users.ts`.
- And we don't want to update the csv whitelist in retrospect.

Reply:
`whitelist_users.ts` is used for whitelisting new users. New users will automatically be given access to default templates (templates available to all/ogp officers/mci officers). The 5th column is used to give access to any additional non-default templates.

`update_whitelist.ts` on the other hand is used in the following scenarios:
1. To give user's added before we implemented template groups access to default templates
2. If we update the list of default templates a group has access to, this script helps to update the permissions in the db for all relevant users.

`update_whitelist.ts` does not take a csv file as input.